### PR TITLE
Use only Docker Machine

### PR DESCRIPTION
**Pull request by devotox**
- Uses only docker machine as we should not let people even use boot2docker
- using -r or --run-command "callback" allows a callback command like "docker-compose up" to be run after and be in the same terminal

### CODE REVIEW

**Comment by brikis98** [comment]
Thanks for the PR! A few requests:
1. There are several unrelated changes here combined into a single PR, which makes it harder to review now and to understand/debug later. Could you create separate PRs for a) removing boot2docker, b) adding the `--run-command`, c) adding support for `config.fish`
2. Please fix the failing tests

Thanks!

**Comment by devotox** [comment]
Alright that sounds good ... I will do that very soon. One thing have you tried this with an Ember app ... for some reason no matter what i do i get these

```
 2016-02-18 17:57:04 [INFO] cannot delete non-empty directory: intranet/tmp/babel-input_base_path-uk7StRBr.tmp
-e 2016-02-18 17:57:04 [INFO] rsync: opendir "/Users/devonte/Downloads/platform/intranet/tmp/babel-input_base_path-tqG6KEEz.tmp" failed: Permission denied (13)
-e 2016-02-18 17:57:04 [INFO] cannot delete non-empty directory: intranet/tmp/babel-input_base_path-tqG6KEEz.tmp
```

My docker compose file has this for the ember app

```
 volumes:
            - /www/platform/intranet/bower_components
            - /www/platform/intranet/node_modules
            - ./intranet:/www/platform/intranet
            - npm_cache:/root/.npm
            - logs:/var/log
            - tmp:/tmp
```

**Comment by brikis98** [comment]
Please file a separate issue for that. I'm guessing the issue is that you need to tell docker-osx-dev to ignore those `/tmp` folders.

**Comment by devotox** [comment]
Before raising it has a problem. I just wanted to make sure I wasn't doing anything wrong. I had the tmp folder in my dockerignore and then the tmp and dist and then also just the dist. No dice

File: README.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ Docker development environment on OS X, including installing Docker and
 Boot2Docker:
 
 ```sh
-curl -o /usr/local/bin/docker-osx-dev https://raw.githubusercontent.com/brikis98/docker-osx-dev/master/src/docker-osx-dev
+curl -o /usr/local/bin/docker-osx-dev https://raw.githubusercontent.com/devotox/docker-osx-dev/master/src/docker-osx-dev
```
Comment (ain) [inline README.md:46]: Thanks for the PR. In addition to what @brikis98 already said, please also keep the official path in the docu.

**Pull request closed without merging**